processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -235,18 +236,27 @@ public SegmentAnalysis apply(@Nullable SegmentAnalysis input)
   @Override
   public <T extends LogicalSegment> List<T> filterSegments(SegmentMetadataQuery query, List<T> segments)
```

Comment:
What's the purpose of the changes in this method? They don't seem related to the rest of the patch.